### PR TITLE
Bitcoin DA tx order

### PR DESCRIPTION
**Pull request by kpp**
# Description

## Linked Issues
- Fixes https://github.com/chainwayxyz/citrea/issues/529

## Testing
`make test`

### CODE REVIEW

**Comment by kpp** [comment]
6d14dc26f46 breaks tests =(

File: crates/bitcoin-da/src/service.rs [hunk 1]
```diff
@@ -424,28 +485,20 @@ impl DaService for BitcoinService {
 
     async fn send_transaction(
         &self,
-        blob: &[u8],
+        _blob: &[u8],
     ) -> Result<<Self as DaService>::TransactionId, Self::Error> {
-        let fee_sat_per_vbyte = self.get_fee_rate().await?;
-        self.send_transaction_with_fee_rate(blob, fee_sat_per_vbyte)
-            .await
+        unimplemented!("Use send_tx_no_wait instead")
```
Comment (eyusufatik) [inline crates/bitcoin-da/src/service.rs:490]: why?


use send_tx_no_wait and wait on the receiver :D

File: crates/bitcoin-da/src/spec/utxo.rs [hunk 2]
```diff
@@ -1,7 +1,7 @@
 use bitcoin::Txid;
 use serde::{Deserialize, Serialize};
 
-#[derive(Clone, Debug, PartialEq, Serialize)]
+#[derive(Clone, Debug, PartialEq, Eq, Serialize, Hash)]
```
Comment (eyusufatik) [inline crates/bitcoin-da/src/spec/utxo.rs:4]: Why is derive Hash is added?

If we need it, we can use txid and vout as it's unique identifier.

**Review by eyusufatik — COMMENTED** [review]
great PR

File: crates/bitcoin-da/src/spec/utxo.rs (hunk 2 above)
Comment (kpp) [reply crates/bitcoin-da/src/spec/utxo.rs:4]: Oh, I was playing with hashmaps. Will fix

File: crates/bitcoin-da/src/service.rs (hunk 1 above)
Comment (kpp) [reply crates/bitcoin-da/src/service.rs:490]: Thats right, one can wait on the receiver anywhere in the code, not to .await at the call cite

**Review by eyusufatik — APPROVED** [review]

**Pull request merged**